keps/sig-node/3542-cri-image-pulling-with-progress-notification/README.md

```diff
@@ -0,0 +1,827 @@
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+In High Performance Computing and other specialized enviroments container
+images often get extremely big. Pulling such images even over high speed
+links can take a lot of time, tens of minutes or more. This often results
+in a bad user experience when deploying a workload triggers pulling such
+an image as a side-effect. For a long time the user has no visibility to
+how creating the workload progresses, or if it progresses at all.
+
+Introducing a new CRI ImageService RPC call for pulling images with a
+streaming progress notification response would provide the low-level
+building blocks to improve this situation.
+
+
+
+
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Extend CRI to provide image pulling progress API that can be utilized by client
+tools (crictl pull), or machinery (kubelet pulling the image)
+
+### Non-Goals
+
+- implementation of utilization of proposed interface additions by the
+client-tools and / or k8s services
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
```

Comment:
This proposal is blank - we'll need this filled in.

If the PR is a work-in-progress, please mark it as a draft @byako.

Reply:
Proposal is now filled, PTAL